dev/package-examples/nginx-1.2.0/dataset/access/manifest.yml

```diff
@@ -8,18 +8,22 @@ type: logs
 # The ingest pipeline which should be used.
 ingest_pipeline: default
 
-vars:
-  - name: paths
-    # Should we define this as array? How will the UI best make sense of it?
-    type: textarea
-    default:
-      - /var/log/nginx/access.log*
-    # I suggest to use ECS fields for this config options here: https://github.com/elastic/ecs/blob/master/schemas/os.yml
-    # This would need to be based on a predefined definition on what can be filtered on
-    os.darwin:
-      - /usr/local/var/log/nginx/access.log*
-    os.windows:
-      - c:/programdata/nginx/logs/*access.log*
+# List of supported inputs
+inputs:
+  - type: log
+    vars:
+      - name: paths
+        required: true
+        # Should we define this as array? How will the UI best make sense of it?
+        type: textarea
+        default:
+          - /var/log/nginx/access.log*
+        # I suggest to use ECS fields for this config options here: https://github.com/elastic/ecs/blob/master/schemas/os.yml
+        # This would need to be based on a predefined definition on what can be filtered on
+        os.darwin:
+          - /usr/local/var/log/nginx/access.log*
+        os.windows:
+          - c:/programdata/nginx/logs/*access.log*
```

Comment:
Yes, good point. They are platform specific defaults. As we should not mix array and object I wonder how the yaml should look like:

        default:
           all:
              - /var/log/nginx/access.log*
        # I suggest to use ECS fields for this config options here: https://github.com/elastic/ecs/blob/master/schemas/os.yml
        # This would need to be based on a predefined definition on what can be filtered on
          os.darwin:
            - /usr/local/var/log/nginx/access.log*
          os.windows:
            - c:/programdata/nginx/logs/*access.log*
```

I don't like the all. Perhaps we find a better name? And should we call it `defaults` instead of `default`?

Reply:
I'm not fond of `all` either, and it seems silly to do `default.default` 🙂 Thinking about this again, what if we just nest `default` under each platform? I don't know if we'll want to add other platform-specific configuration for vars, but this schema could support that too since it essentially mirrors all the top-level vars properties. As an example, maybe `paths` is not required for `windows` for whatever reason:

```yml
        default:
            - /var/log/nginx/access.log*
        os.darwin:
          default:
            - /usr/local/var/log/nginx/access.log*
        os.windows:
          required: false
          default:
            - c:/programdata/nginx/logs/*access.log*
```

Reply:
LGTM. So in summary: All configs from the top level could also be nested under each platform. Lets try it.

What I suggest for now, lets ignore platform at least on the UI side.